src/gamestate/system_state.cpp

```diff
@@ -3898,6 +3898,15 @@ void state::single_game_tick() {
 			ai::refresh_home_ports(*this);
 			break;
 		case 1:
+			// Instant research cheat
+			if(this->user_settings.instant_research_enabled) {
+				auto n = this->local_player_nation;
```

Comment:
I believe that this would instantly de-sync multiplayer, as each instance of the game has a different local player set and so would end up with different gamestates after this step. Now, I don't know if leaf designed the cheats to work in multiplayer. If they are blocked from working in multiplayer, than this is fine, but if they can be used in multiplayer, then some other approach will be needed.